src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1345,7 +1345,7 @@ private Result AfterLeftChildOfBinaryOperatorHasBeenVisited(BoundBinaryOperator
 
                 Debug.Assert(!IsConditionalState);
                 var result = Result.Create(InferResultNullability(binary, leftType, rightType));
```

Comment:
Do we still need local `result`?

Reply:
`result` is used as the return value later in the method. It may not be necessary (assuming the visitor state does not change before that point), but it seems useful just in case.